va/va.go

```diff
@@ -707,6 +720,14 @@ func (va *ValidationAuthorityImpl) DoDCV(ctx context.Context, req *vapb.PerformV
 	}()
 
 	// Do local validation. Note that we process the result in a couple ways
+	// For dns-account-01 challenges, construct the account URI from the configured prefix
+	var accountURI string
+	if chall.Type == core.ChallengeTypeDNSAccount01 && features.Get().DNSAccount01Enabled {
+		if va.dnsAccountChallengeURIPrefix != "" && req.Authz.RegID != 0 {
```

Comment:
This conditional is IMO semantically broken unless something upstream of it -- main(), or NewValidationAuthorityImpl -- enforces that dnsAccountChallengeURIPrefix _must_ be set if the DNSAccount01Enabled feature flag is set. It should be impossible to proceed to validation with accountURI being the empty string.

I don't want to be in a situation where we're failing Subscribers' dns-account-01 validations because we forgot to configure a URI prefix. We need to be failing at startup time, instead.